src/main/java/com/elasticbox/jenkins/model/repository/api/WorkspacesRepositoryApiImpl.java

```diff
@@ -72,6 +72,9 @@ public boolean apply(AbstractWorkspace it) {
         } catch (IOException e) {
             logger.log(Level.SEVERE, "There is no workspaces");
             throw new RepositoryException("There is no workspaces");
+        } catch (NullPointerException npe) {
```

Comment:
Yes, it is because of that... but it is hard to control before, because it is called from another constructor that can be called from many places, and in some cases client is cached, in some others is not initialized (yet), but eventually will be regenerated with a valid client, so I opted for just controlling the exception when it is not valid, because another object will be generated with a valid client when all data is available (in the Jenkins config form).